backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/StatementRecorder.java

```diff
@@ -0,0 +1,46 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.cassandra;
+
+
+import java.util.List;
+
+import com.datastax.driver.core.Statement;
+import com.google.common.collect.ImmutableList;
+
+public class StatementRecorder {
+    private final ImmutableList.Builder<Statement> statements;
+
+    public StatementRecorder() {
+        statements = ImmutableList.builder();
+    }
+
+    void recordStatement(Statement statement) {
+        synchronized (statements) {
```

Comment:
I would rather use a concurrent datastructure that doesn't require a lock to avoid introducing contention on the recorder (and thus have a different execution pattern than in production)